### PR TITLE
AROR-1653 Localytics ODBC adapter needs updated to Rails 5

### PR DESCRIPTION
**Pull request by schallereqo**

### CODE REVIEW

File: lib/active_record/connection_adapters/odbc_adapter.rb [hunk 1]
```diff
@@ -173,7 +174,10 @@ def translate_exception(exception, message)
 
         case
         when error_number == ERR_DUPLICATE_KEY_VALUE
-          ActiveRecord::RecordNotUnique.new(message, exception)
+          args = [message]
+          args << exception if ActiveRecord::RecordNotUnique.instance_method(:initialize).arity == 2
```
Comment (johvet) [inline lib/active_record/connection_adapters/odbc_adapter.rb:178]: 👍

File: test/connections_test.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  def test_reconnect!
+    @old_raw_connection = @connection.raw_connection
+    assert_equal true, @connection.active?
+    @connection.reconnect!
+    refute_equal @old_raw_connection, @connection.raw_connection
+    assert_equal true, @connection.active?
+  end
+end
```
Comment (johvet) [inline test/connections_test.rb]: the EOF seems wrong

**Review by johvet — CHANGES_REQUESTED** [review]
This looks all good to me, except for a single EOF that needs to be addressed.

**Review by johvet — APPROVED** [review]

**Review by ziggythehamster — APPROVED** [review]

**Comment by schallereqo** [comment]
I will merge this and then go update the projects that are using the gem

**Pull request merged**